asv/_rangemedian.cpp

```diff
@@ -348,52 +348,14 @@ static PyMethodDef RangeMedian_methods[] = {
 
 static PyTypeObject RangeMedianType = {
     PyVarObject_HEAD_INIT(NULL, 0)
-    "RangeMedian",
-    sizeof(RangeMedianObject),
-    0,
-    (destructor)RangeMedian_dealloc, // tp_dealloc
-    0,                          // tp_print
-    0,                          // tp_getattr
-    0,                          // tp_setattr
-    0,                          // tp_compare / tp_reserved
-    0,                          // tp_repr
-    0,                          // tp_as_number
-    0,                          // tp_as_sequence
-    0,                          // tp_as_mapping
-    0,                          // tp_hash
-    0,                          // tp_call
-    0,                          // tp_str
-    0,                          // tp_getattro
-    0,                          // tp_setattro
-    0,                          // tp_as_buffer
-    Py_TPFLAGS_DEFAULT,         // tp_flags
-    NULL,                       // tp_doc
-    0,                          // tp_traverse
-    0,                          // tp_clear
-    0,                          // tp_richcompare
-    0,                          // tp_weaklistoffset
-    0,                          // tp_iter
-    0,                          // tp_iternext
-    RangeMedian_methods,        // tp_methods
-    0,                          // tp_members
-    0,                          // tp_getset
-    0,                          // tp_base
-    0,                          // tp_dict
-    0,                          // tp_descr_get
-    0,                          // tp_descr_set
-    0,                          // tp_dictoffset
-    (initproc)RangeMedian_init, // tp_init
-    0,                          // tp_alloc
-    RangeMedian_new,            // tp_new
-    0,                          // tp_free
-    0,                          // tp_is_gc
-    0,                          // tp_bases
-    0,                          // tp_mro
-    0,                          // tp_cache
-    0,                          // tp_subclasses
-    0,                          // tp_weaklist
-    0,                          // tp_del
-    0,                          // tp_version_tag
+    .tp_name="RangeMedian",
+    .tp_basicsize=sizeof(RangeMedianObject),
+    .tp_dealloc=(destructor)RangeMedian_dealloc,
+    .tp_flags=Py_TPFLAGS_DEFAULT,
+    .tp_doc = NULL,
+    .tp_methods=RangeMedian_methods,
+    .tp_init=(initproc)RangeMedian_init,
+    .tp_new=RangeMedian_new
```

Comment:
Over [at cython](https://github.com/cython/cython/pull/5435#issuecomment-2090608598), using this in c++ requires c20, so it errored out on windows with `error C7555: use of designated initializers requires at least '/std:c++20'`. Does it work here?

Reply:
Designated initializers is a [C++20 feature](https://en.cppreference.com/w/cpp/language/aggregate_initialization) but also [C99 feature](https://rgambord.github.io/c99-doc/sections/6/7/8/index.html#p34), so here this should be fine (this is also now the [recommended style](https://docs.python.org/3/c-api/typeobj.html#examples) in the Python documentation). Although in the rewrite it would be much much better to cleanly separate the C++ parts (`vector` etc) from the Python bindings `C`.

Reply:
Actually thinking about this a bit more I'm just going to revert the change here, there's a lot of weirdness here including the use of a template in a `.cpp` :'D